src/celeritas/em/interactor/WentzelInteractor.hh

```diff
@@ -0,0 +1,172 @@
+    inline CELER_FUNCTION WentzelInteractor(WentzelRef const& shared,
+                                            ParticleTrackView const& particle,
+                                            Real3 const& inc_direction,
+                                            MaterialView const& material,
+                                            ElementComponentId const& elcomp_id,
+                                            CutoffView const& cutoffs);
+
+    //! Sample an interaction with the given RNG
+    template<class Engine>
+    inline CELER_FUNCTION Interaction operator()(Engine& rng);
+
+  private:
+    //// DATA ////
+
+    // Constant shared data
+    WentzelRef const& data_;
+
+    // Incident direction
+    Real3 const& inc_direction_;
+
+    // Incident particle
+    ParticleTrackView const& particle_;
+
+    // Mott coefficients of the target element
+    WentzelElementData const& element_data_;
+
+    // Material's cutoff energy for the incident particle
+    real_type const cutoff_energy_;
+
+    // Target element
+    ElementView const element_;
+
+    //// HELPER FUNCTIONS ////
+
+    //! Calculates the recoil energy for the given scattering direction
+    inline CELER_FUNCTION real_type calc_recoil_energy(
+        Real3 const& new_direction, Mass const& target_mass) const;
+};
+
+//---------------------------------------------------------------------------//
+// INLINE DEFINITIONS
+//---------------------------------------------------------------------------//
+/*!
+ * Construct from shared and state data
+ */
+CELER_FUNCTION
+WentzelInteractor::WentzelInteractor(WentzelRef const& shared,
+                                     ParticleTrackView const& particle,
+                                     Real3 const& inc_direction,
+                                     MaterialView const& material,
+                                     ElementComponentId const& elcomp_id,
+                                     CutoffView const& cutoffs)
+    : data_(shared)
+    , inc_direction_(inc_direction)
+    , particle_(particle)
+    , element_data_(shared.elem_data[material.element_id(elcomp_id)])
+    , cutoff_energy_(value_as<Energy>(cutoffs.energy(particle.particle_id())))
+    , element_(material.make_element_view(elcomp_id))
+{
+    CELER_EXPECT(particle_.particle_id() == data_.ids.electron
+                 || particle_.particle_id() == data_.ids.positron);
+    CELER_EXPECT(particle_.energy() > detail::coulomb_scattering_limit()
+                 && particle_.energy() < detail::high_energy_limit());
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Sample the Coulomb scattering of the incident particle.
+ */
+template<class Engine>
+CELER_FUNCTION Interaction WentzelInteractor::operator()(Engine& rng)
+{
+    // Select an isotope of the target nucleus
+    IsotopeSelector iso_select(element_);
+    IsotopeView target = element_.make_isotope_view(iso_select(rng));
```

Comment:
Since this interaction is with an isotope rather than an element, would it make sense for the isotope to be sampled outside the `Interactor` (in the `Executor`)? (I'm not deep enough into reviewing this to answer myself, just something for you to consider.)